### DISCUSSION TITLE
Getting an accurate framerate from ImGui::IO

### DISCUSSION THREAD

**Discussion by Nemirtingas** [ocornut/imgui#4138] [Old/Archived discussions]
Hi,
I'm quite puzzled actually. I was making a perf counter window, I added a fps counter as well as a min/max fps.
I used `ImGui::GetIO().Framerate` to get the immediate framerate, there is a big spike at start and it mess my stats because I also capture the best/lowest fps of the application lifetime. I think it is caused by `ImGui_ImplWin32_Init`, `g_Time` is initialized and right after that, there is a `ImGui_ImplWin32_NewFrame` called making Dear ImGui computes some wrong framerate. With the current Dear ImGui code, my max constant framerate is 120fps, never higher.

Then I removed the `g_time` initialization, that way, the first frames are near 0, but by doing so, my max constant framerate is always around 122-125fps.

I can reproduce this everytime, any idea on how initializing g_time to 0 instead of the current time makes the max framerate caps at 122 fps instead of 120fps ?

https://imgur.com/a/FSN5iyF

**Comment by ocornut** [comment]
> I used ImGui::GetIO().Framerate to get the immediate framerate

This is not the "immediate/current" framerate, it is the rolling average framerate over the last 120 frames.
If you want each frame time use 1.0f / io.DeltaTime or compute it your own way.

However we computed the rolling average wrongly for the first 120 frames. I improved the code now: 
https://github.com/ocornut/imgui/commit/41030cbfe2a4cbce186a5cacc726a0b2378b15d7

But don't expect this to be very useful for your purpose, and don't expect initial first few frames of your app to be meaningful as many components and systems will be warming up.

**Reply by Nemirtingas** [reply]
Yep, I did understand perfectly that this was an estimation of the framerate and that it also was an average. But I didn't understand why having the initial gTime set to 0 was making the fps go up to 122~125fps, not only for the first 120 frames, and having it set to the current function time caps it at 120fps (120 is my screen refresh rate and v-sync on so it is quite accurate).

I'm not making some high precision fps count, its just some fancy stuff I want to add to my in game overlay, I'm also making an average over 1 second long and displaying it.
